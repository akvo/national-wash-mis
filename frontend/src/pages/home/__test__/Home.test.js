import { render, screen } from "@testing-library/react";
import TestApp from "../../../TestApp";
import "@testing-library/jest-dom";

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    };
  };

jest.mock("axios");
jest.mock("leaflet");

describe("Home page", () => {
  const { asFragment } = render(<TestApp />);
  test("test if About RUSH exists", () => {
    expect(screen.getByText("About RUSH")).toBeInTheDocument();
    expect(
      screen.getByText(
        "The Kenya Rural Urban Sanitation and Hygiene (RUSH) platform is a real-time monitoring and information system owned by the Ministry of Health. The platform aggregates quantitative and qualitative data from county and national levels and facilitates data analysis, report generation and visualizations."
      )
    ).toBeInTheDocument();
    expect(asFragment).toMatchSnapshot();
  });
});
