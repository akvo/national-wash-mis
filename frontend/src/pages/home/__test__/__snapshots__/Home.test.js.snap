// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page test if About RUSH exists 1`] = `[Function]`;
